src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts

```diff
@@ -293,13 +293,16 @@ export class ReadingsGraphComponent implements OnDestroy {
     const graphStartTimeSeconds = this.payload.start === 0 ? (now - this.payload.len) : this.payload.start;
     const graphStartDateTime = moment(graphStartTimeSeconds * 1000).format('YYYY-M-D H:mm:ss.SSS');
     this.layout.xaxis['range'] = [graphStartDateTime, timestamps[0]];
+    const timeWindow = Utils.getTimeWindow(this.timeWindowIndex);
+    this.updateXAxisTickFormat(timeWindow.value);
   }
 
   public zoomGraph(seconds: number) {
     this.zoom = true;
     const maxDataPoints = 600;
     const bucket = seconds / maxDataPoints;
     const length = seconds;
+    this.updateXAxisTickFormat(length);
```

Comment:
when there are many readings in DB, API response may be delayed. do we want to update here (per timeWindow text) or after `getAssetReadings` call.

Reply:
Removed from here, now it is updated after `getAssetReadings` call.